source/particle/property/cpo_elastic_tensor.cc

```diff
@@ -0,0 +1,582 @@
+          sqrt(2)*0.25*(input_tensor[1][2][0][2]+input_tensor[1][2][2][0]+input_tensor[2][1][0][2]+input_tensor[2][1][2][0]+input_tensor[0][2][1][2]+input_tensor[0][2][2][1]+input_tensor[2][0][1][2]+input_tensor[2][0][2][1])  // 20 // 21
+        });
+
+      }
+
+      template <int dim>
+      UpdateTimeFlags
+      CpoElasticTensor<dim>::need_update() const
+      {
+        return update_output_step;
+      }
+
+      template <int dim>
+      UpdateFlags
+      CpoElasticTensor<dim>::get_needed_update_flags () const
+      {
+        return update_default;
+      }
+
+      template <int dim>
+      std::vector<std::pair<std::string, unsigned int>>
+      CpoElasticTensor<dim>::get_property_information() const
+      {
+        std::vector<std::pair<std::string,unsigned int>> property_information;
+
+        property_information.push_back(std::make_pair("cpo_elastic_tensor",SymmetricTensor<2,6>::n_independent_components));
+
+        return property_information;
+      }
+
+      template <int dim>
+      void
+      CpoElasticTensor<dim>::declare_parameters (ParameterHandler &prm)
+      {
+        prm.enter_subsection("Postprocess");
+        {
+          prm.enter_subsection("Particles");
+          {
+            prm.enter_subsection("Crystal Preferred Orientation");
+            {
+              prm.declare_entry ("Number of grains per particle", "50",
+                                 Patterns::Integer (1),
+                                 "The number of grains of each different mineral "
+                                 "each particle contains.");
+
+              prm.enter_subsection("Initial grains");
+              {
+                prm.enter_subsection("Uniform grains and random uniform rotations");
+                {
+                  prm.declare_entry ("Minerals", "Olivine: Karato 2008, Enstatite",
+                                     Patterns::List(Patterns::Anything()),
+                                     "This determines what minerals and fabrics or fabric selectors are used used for the LPO calculation. "
+                                     "The options are Olivine: Passive, A-fabric, Olivine: B-fabric, Olivine: C-fabric, Olivine: D-fabric, "
+                                     "Olivine: E-fabric, Olivine: Karato 2008 or Enstatite. Passive sets all RRSS entries to the maximum. The "
+                                     "Karato 2008 selector selects a fabric based on stress and water content as defined in "
+                                     "figure 4 of the Karato 2008 review paper (doi: 10.1146/annurev.earth.36.031207.124120).");
+                }
+                prm.leave_subsection ();
+              }
+              prm.leave_subsection ();
+            }
+            prm.leave_subsection ();
+          }
+          prm.leave_subsection ();
+        }
+        prm.leave_subsection ();
+      }
```

Comment:
Arent all of these parameters already declared in the main  CPO plugin? No need to declare them again here. Just keep the parse_parameters function and leave a comment that describes where the parameters have been declared. Using only a single declaration is safer in case we decide to rename the variables.